### ISSUE TITLE
Windows 10_Getting an error "GPU process launch failed: error_code=18" while trying to open cypress test runner.

### ISSUE THREAD

**Issue by Praveentrp** [cypress-io/cypress#19924]
### Current behavior

**Console log:**
`Note: You have set the environment variable:

CYPRESS_RUN_BINARY=C:\Cypress\9.3.1\Cypress\Cypress.exe

This overrides the default Cypress binary path used.

Found binary version 9.1.0 installed in C:\Cypress\9.3.1\Cypress`

!! Warning: Binary version 9.1.0 does not match the expected package version 9.3.1

 These versions may not work properly together.

It looks like this is your first time using Cypress: 9.1.0

  Verified Cypress! C:\Cypress\9.3.1\Cypress

Opening Cypress...

[22552:0126/205620.709:ERROR:gpu_process_host.cc(951)] GPU process launch failed: error_code=18
[22552:0126/205620.725:ERROR:gpu_process_host.cc(951)] GPU process launch failed: error_code=18
[22552:0126/205620.742:ERROR:gpu_process_host.cc(951)] GPU process launch failed: error_code=18
[22552:0126/205620.758:ERROR:gpu_process_host.cc(951)] GPU process launch failed: error_code=18
[22552:0126/205620.772:ERROR:gpu_process_host.cc(951)] GPU process launch failed: error_code=18
[22552:0126/205620.799:ERROR:gpu_process_host.cc(951)] GPU process launch failed: error_code=18
[22552:0126/205620.712:ERROR:gpu_process_host.cc(951)] GPU process launch failed: error_code=18
[22552:0126/205620.788:ERROR:gpu_process_host.cc(951)] GPU process launch failed: error_code=18
[22552:0126/205620.793:ERROR:gpu_process_host.cc(951)] GPU process launch failed: error_code=18
[22552:0126/205628.793:FATAL:gpu_data_manager_impl_private.cc(417)] GPU process isn't usable. Goodbye.

### Desired behavior

Cypress test runner should be successfully open.

### Test code to reproduce

1. To install the cypress latest version using the below command. (As GitHub has been blocked on-premise, we are using the below  command for the installation)
CYPRESS_INSTALL_BINARY='C:/CypressInstallables/cypress-win-9.1.0.zip' CYPRESS_CACHE_FOLDER='C:/Cypress' npm i cypress --sev-dev

2. Cypress 9.3.1 got successfully installed.

3. Environment variable has been set in the System variables.
CYPRESS_RUN_BINARY=C:\Cypress\9.3.1\Cypress\Cypress.exe

4. npx cypress open

### Cypress Version

9.3.1

**Comment by pratr5** [comment]
Hi @jennifer-shehane  - Could you please look into the issue.

**Comment by tbiethman** [comment]
Maybe related: https://github.com/electron/electron/issues/32074

**Comment by Praveentrp** [comment]
Hi @tbiethman - Thanks for looking into it. The impacted number of users is pretty high on respective to this issue in Windows 10 machine. Kindly look into this issue as a priority and provide a resolution on the same.

**Comment by flotwig** [comment]
@Praveentrp Can you try launching the app with the environment variable `ELECTRON_EXTRA_LAUNCH_ARGS=--disable-gpu-sandbox`?

**Comment by Praveentrp** [comment]
Thanks, @flotwig for your reply.

Cypress test runner is successfully getting launched after adding this environment variable "ELECTRON_EXTRA_LAUNCH_ARGS=--disable-gpu-sandbox". I am not getting this error "GPU process launch failed:" now.

But while trying to run the customized execution command, I am getting the below error.

`Still waiting to connect to chrome, retrying in 1 second (attempt 60/62)
Still waiting to connect to chrome, retrying in 1 second (attempt 61/62)
Still waiting to connect to chrome, retrying in 1 second (attempt 62/62)
Cypress failed to make a connection to the Chrome DevTools Protocol after retrying for 50 seconds.

This usually indicates there was a problem opening the chrome browser.`

Could you please help us to resolve this issue as well?

**Comment by narayananpalani** [comment]
As @Praveentrp mentioned, it is also affecting Win10 of both Microsoft Surface Laptops and HP EliteBooks across cypress user community.

**Comment by FlorianeTy** [comment]
Hi, we have the same problem in our company and adding environment variable doesn't correcting the problem.
We're working on HP Elitebook and Windows 10, about ten employees are impacted in my department.

![image](https://user-images.githubusercontent.com/15136803/157420960-82dac624-b072-453a-935e-723d049501ab.png)

Could you please investigate the issue ?

**Comment by nunorbatista** [comment]
any progress? I'm seeing the same error

**Comment by cezarsmpio** [comment]
Same here after resolving `X display` issues.

**Comment by Brunight** [comment]
Same here, running in a Ubuntu WSL 2 instance inside Windows 11

**Comment by jakohans76** [comment]
I have the same issue on my Win10. I can open Cypress with v8.5.0 but it crashes with above error message since v8.6.0.

**Comment by SaajidNawwar** [comment]
I'm having the same issue. I get 'GPU process exited unexpectedly: exit_code=34'

**Comment by Praveentrp** [comment]
**Solution:**

1. Try opening your IDE/git bash in non-admin mode.
2. Run the command "./node_modules./bin/cypress open"
3. Click on the feature file which you want to execute.

This has resolved my issue. Hope it will resolve for you guys as well.

**Comment by SaajidNawwar** [comment]
@Praveentrp I tried executing it without admin mode yet I got the same error.

**Comment by Pradyyy-rgb** [comment]
Hello everyone, I have been facing the exact same issue as mentioned above,

Spec:  Laptop:Microsoft surface and with every browser it has shown different kind of error,
          Cypress 9.5.3 installed not from web
          **Env variable**: Name: ELECTRON_EXTRA_LAUNCH_ARGS, Value: --disable-gpu-sandbox npx cypress run
                                     Name: NVM_HOME, Value: c:/users/username/AppData/Roaming/nvm
         **System variable**: HTTP_Proxy, value: 'proxy set' with port 8080

When I open the runner with `./node_modules./bin/cypress open` on an admin git bash it open the runner successfully
Then I select my browser as chrome
And I click on the action.spec.js 

Actual behavior: 
![image](https://user-images.githubusercontent.com/82174459/203356620-2e5b1b3c-933d-4e82-bf1d-b43799182ada.png)

![image](https://user-images.githubusercontent.com/82174459/203357576-522b0f7f-c677-4121-82f0-96af1407d1c9.png)

same error when running with Edge
Then opened with Electron

error: 
![image](https://user-images.githubusercontent.com/82174459/203359422-559f0cc1-f3ba-430d-a3e7-b1960e0e74c6.png)


When I run it with `npx cypress open` from the admin terminal , It throws 
![Uploading image.png…]()

I'm not allowed to run it as non admin

**Comment by JonathanFL** [comment]
When I installed Cypress globally on my PC and opened PowerShell (not as admin) in the project directory and ran `cypress open` from there - then it worked. Hope it also works for others :)

**Comment by nagash77** [comment]
This issue has gone stale and we do not have a way to reproduce this specific behavior at this point.  I am going to close this issue for the time being and if a reproducible example or new information comes to light please comment back here and we will see if we can make further progress.

**Comment by pebcakerror** [comment]
just tried cypress for the first time today. 

Installed via `npm install cypress --save-dev` and was getting the `GPU process launch failed`. 

I tried via windows terminal, non-admin powershell, admin powershell, vs code git gui. all had the same error.

When installed as a global `npm install -g cypress` then running `cypress open` it was able to run.

However, n0ow there is a new error from
```
Circular symlink detected: "C:\Users\me\AppData\Local\Application Data" points to "C:\Users\me\AppData\Local"
Error: Circular symlink detected: "C:\Users\me\AppData\Local\Application Data" points to "C:\Users\me\AppData\Local"
    at E._getEntryType (<embedded>:4141:7225)
    at async E._read (<embedded>:4141:5965)
````

Node 18.17.1 installed via nvm windows

**Comment by MikeMcC399** [comment]
@pebcakerror 

Can you confirm that you are getting exactly the error message
"GPU process launch failed: error_code=18"
including the error code `18`?

Have you checked if there is an updated graphics driver available for your system?

Which version of Windows are you running?
Are you running in a virtual environment, or directly in Windows?
Are you running on a mapped drive or Microsoft OneDrive?

I tried installing Cypress globally, like you, on Windows 11, and I did not get any symlink error.

**Comment by Christoffer92** [comment]
My issue looks alot like the other users here.

I also just tried Cypress for the first time today.

Installed via npm install cypress --save-dev and was getting the GPU process launch failed.

Because of group policies i can only run the application in powershell (Admin)

@MikeMcC399 Here is a screenshot confirming its the error code 18

![Image](https://github.com/user-attachments/assets/56373e50-0508-4d66-ba9e-1f8da8bcebee)

**Comment by MikeMcC399** [comment]
@Christoffer92 

This issue was opened for an older version of Cypress and it is now closed. You can open a new issue for your problem.

If you are attempting to use `ELECTRON_EXTRA_LAUNCH_ARGS`, then you would need to spell it correctly.

- Edit: opened in #31659
